src/v2/util/BaseLoginRouter.js

```diff
@@ -120,44 +113,30 @@ function (Okta, BrowserFeatures, Settings,
       });
 
       this.listenTo(this.appState, 'change:remediationFailure', function (appState, err) {
-        RouterUtil.routeAfterAuthStatusChangeError(this, err);
+        // Global error handling for CORS enabled errors
+        if (err.xhr && BrowserFeatures.corsIsNotEnabled(err.xhr)) {
+          this.settings.callGlobalError(new Errors.UnsupportedBrowserError(
+            Okta.loc('error.enabled.cors')
+          ));
+          return;
+        }
+        //set flashError
+        this.appState.set('flashError', err);
+        this.defaultAuth();
       });
 
       this.listenTo(this.appState, 'change:remediationSuccess', function (appState, trans) {
-        RouterUtil.routeAfterAuthRemediationChange(this, trans.data);
+        // set authStateRefreshed
+        this.settings.set('authStateRefreshed', true);
```

Comment:
For cases when we come to the router without a stateToken, we need to be able to tell if it is because we did an introspect call and then unset the stateToken or widget was bootstrapped without a stateToken. So if there was a stateToken and we hit introspect we use `authStateRefreshed ` to route accordingly. `remediationSuccess ` would also have the response, but we would have to check for that `remediationSuccess ` object being non empty. I preferred checking for a  boolean in `authStateRefreshed`.

Reply:
> but we would have to check for that remediationSuccess object being non empty.